pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -373,131 +396,231 @@ func (p *Plugin) scheduleDeleteTask(ctx context.Context) {
+			AliasName:   &entry.AliasName,
+			TargetKeyId: &entry.Arn,
+		})
+		if err != nil {
+			p.log.Error("It was not possible to refresh alias",
+				aliasNameTag, entry.AliasName, keyArnTag, entry.Arn, reasonTag, err)
+		}
 	}
 }
 
-type keyFetcher struct {
-	log       hclog.Logger
-	kmsClient kmsClient
-	keyPrefix string
+func (p *Plugin) disposeAliasesTask(ctx context.Context) {
+	ticker := p.clk.Ticker(disposeAliasesFrequency)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Debug("Stopping dispose aliases task", reasonTag, ctx.Err())
+			return
+		case <-ticker.C:
+			p.disposeAliases(ctx)
+		}
+	}
 }
 
-func (kf *keyFetcher) fetchKeyEntries(ctx context.Context) ([]*keyEntry, error) {
-	var keyEntries []*keyEntry
-	var keyEntriesMutex sync.Mutex
-	paginator := kms.NewListAliasesPaginator(kf.kmsClient, &kms.ListAliasesInput{Limit: aws.Int32(100)})
-	g, ctx := errgroup.WithContext(ctx)
+func (p *Plugin) disposeAliases(ctx context.Context) {
+	p.log.Debug("Looking for aliases in trust domain to dispose")
+	paginator := kms.NewListAliasesPaginator(p.kmsClient, &kms.ListAliasesInput{Limit: aws.Int32(100)})
 
 	for {
 		aliasesResp, err := paginator.NextPage(ctx)
-		if err != nil {
-			return nil, newErrorf(codes.Internal, "failed to fetch aliases: %v", err)
+		switch {
+		case err != nil:
+			p.log.Error("Failed to fetch aliases to dispose", reasonTag, err)
+			return
+		case aliasesResp == nil:
+			p.log.Error("Failed to fetch aliases to dispose: nil response")
+			return
 		}
-		kf.log.Debug("Found aliases", "num_aliases", len(aliasesResp.Aliases))
 
 		for _, alias := range aliasesResp.Aliases {
 			switch {
-			case alias.TargetKeyId == nil:
-				// this means something external to the plugin created the alias, without associating it to a key.
-				// it should never happen with CMKs.
-				return nil, newErrorf(codes.FailedPrecondition, "failed to fetch aliases: found SPIRE alias without key: %q", *alias.AliasArn)
-			case alias.AliasArn == nil:
-				return nil, newErrorf(codes.Internal, "failed to fetch aliases: found SPIRE alias without arn: %q", *alias.AliasArn)
-			case alias.AliasName == nil:
-				return nil, newError(codes.Internal, "failed to fetch aliases: found alias without a name")
+			case alias.AliasName == nil || alias.LastUpdatedDate == nil || alias.AliasArn == nil:
+				continue
+				// if alias does not belong to trust domain skip
+			case !strings.HasPrefix(*alias.AliasName, p.aliasPrefixForTrustDomain()):
+				continue
+			// if alias belongs to current server skip
+			case strings.HasPrefix(*alias.AliasName, p.aliasPrefixForServer()):
+				continue
 			}
 
-			spireKeyID, ok := kf.spireKeyIDFromAlias(*alias.AliasName)
-			// ignore aliases/keys not belonging to this server
-			if !ok {
+			now := time.Now()
+			diff := now.Sub(*alias.LastUpdatedDate)
+			// diff should never be negative, just being extra careful
+			if diff < 0 || diff < aliasThreshold {
```

Comment:
Doesn't `diff < aliasThreshold` also cover the `diff < 0` case?

Reply:
You are correct! I will fix it.